marklogic-data-hub-central/ui/src/components/modeling/entity-type-modal/entity-type-modal.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState, useEffect, useContext } from 'react';
+import { Modal, Form, Input, Tooltip } from 'antd';
+import { faQuestionCircle } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import styles from './entity-type-modal.module.scss'
+
+import { UserContext } from '../../../util/user-context';
+import { createModel } from '../../../api/modeling';
+import { ModelingTooltips } from '../../../config/tooltips.config';
+
+type Props = {
+  isVisible: boolean;
+  toggleModal: (isVisible: boolean) => void;
+  newEntityAdded: () => void;
+};
+
+const EntityTypeModal: React.FC<Props> = (props) => {
+  const { handleError, resetSessionTime } = useContext(UserContext);
+  // Must start with a letter & only letters, numbers, hyphens, and underscores are allowed
+  const NAME_REGEX = new RegExp('^[A-Za-z][A-Za-z0-9_-]*$');
+  const formItemLayout = {
+    labelCol: { span: 4},
+    wrapperCol: { span: 18 },
+  };
+
+  const [name, setName] = useState('');
+  const [isNameValid, toggleIsNameValid] = useState(true);
+  const [description, setDescription] = useState('');
+
+  useEffect(() => {
+    if (props.isVisible){
+      setName('')
+      setDescription('')
+    }
+  }, [props.isVisible]);
+
+  const handleChange = (event) => {
+    if (event.target.id === 'name') {
+      setName(event.target.value);
+    }
+    if (event.target.id === 'description') {
+      setDescription(event.target.value);
+    }
+  }
+
+  const onOk = (event) => {
+    event.preventDefault();
+    if (NAME_REGEX.test(name)) {
+      toggleIsNameValid(true);
+      createEntityType(name, description);
+    } else {
+      toggleIsNameValid(false);
+    }
+  };
+
+  const createEntityType = async (entityName: string, description: string) => {
+    try {
+      const response = await createModel(entityName, description);
+      if (response['data']) {
+        props.newEntityAdded();
+      }
+    } catch (error) {
+      handleError(error);
+    } finally {
+      resetSessionTime();
+    }
+  }
+
+  const onCancel = () => {
+    props.toggleModal(false)
+  };
+
+  return (
+    <Modal
+      visible={props.isVisible} 
+      closable={true}
+      title={"Add Entity Type"} 
+      cancelText="Cancel"
+      onCancel={() => onCancel()} 
+      okText="Add"
+      onOk={onOk}
+      okButtonProps={{form:'entity-type-form', htmlType: 'submit'}}
+      maskClosable={false}
+      data-testid="entity-type-modal"
```

Comment:
I would also like us to avoid using data-testid heavily and use it as a last resort. We can instead use getByTitle, getByLabelText, getByPlaceholderText etc wherever possible.